Recruit-Training/Advanced-Recruit-Training/Telemetry-Teaser/src/app/app.component.html

```diff
@@ -1,18 +1,36 @@
 <body>
   <div>
     <h1> Elysia Range Simulator </h1>
-    <form name="simulator" #simulator="ngForm" novalidate>
+    <form [formGroup]="rangeForm" name="simulator" #simulator="ngForm" novalidate>
       <label>Speed (km/h):</label>
       <br>
-      <input name="speed" type="number" placeholder="Speed" (keyup)="onKeySpeed($event)">
+      <input class = "form-control" formControlName = "speedInput" name="speed" type="number" placeholder="Speed" (keyup)="onKeySpeed($event)">
+      <br>
+      <div *ngIf="speedInput.invalid">
+        <label *ngIf="speedInput.errors.required" [ngClass] = "'error'">Speed is required</label>
+        <label *ngIf="speedInput.errors.min" [ngClass] = "'error'">The speed should be with the range of 0 to 90</label>
+        <label *ngIf="speedInput.errors.max" [ngClass] = "'error'">The speed should be with the range of 0 to 90</label>
+      </div>
       <br>
       <label>Battery Percentage (%):</label>
       <br>
-      <input name="battery"type="number" placeholder="Battery" (keyup)="onKeyBattery($event)">
+      <input class = "form-control" formControlName = "batteryInput" name="battery"type="number" placeholder="Battery" (keyup)="onKeyBattery($event)">
+      <br>
+      <div *ngIf="batteryInput.invalid">
+        <label *ngIf="batteryInput.errors.required" [ngClass] = "'error'">Battery is required</label>
```

Comment:
Instead of the error label saying 'Battery is required', can the label say 'Battery percentage is required'